test/parallel/test-vm-module-referrer-realm.js

```diff
@@ -0,0 +1,72 @@
+// Flags: --experimental-vm-modules
+'use strict';
+const common = require('../common');
+
+const assert = require('assert');
+const { Script, SourceTextModule, createContext } = require('vm');
+
+async function test() {
+  const foo = new SourceTextModule('export const a = 1;');
+  await foo.link(common.mustNotCall());
+  await foo.evaluate();
+
+  const ctx = createContext({}, {
+    importModuleDynamically: common.mustCall((specifier, wrap) => {
+      assert.strictEqual(specifier, 'foo');
+      assert.strictEqual(wrap, ctx);
+      return foo;
+    }, 2),
+  });
+  {
+    const s = new Script('Promise.resolve("import(\'foo\')").then(eval)', {
+      importModuleDynamically: common.mustNotCall(),
+    });
+
+    const result = s.runInContext(ctx);
+    assert.strictEqual(foo.namespace, await result);
+  }
+
+  {
+    const m = new SourceTextModule('globalThis.fooResult = Promise.resolve("import(\'foo\')").then(eval)', {
+      context: ctx,
+      importModuleDynamically: common.mustNotCall(),
+    });
+    await m.link(common.mustNotCall());
+    await m.evaluate();
+    assert.strictEqual(foo.namespace, await ctx.fooResult);
+    delete ctx.fooResult;
+  }
+}
+
+async function testMissing() {
+  const ctx = createContext({});
+  {
+    const s = new Script('Promise.resolve("import(\'foo\')").then(eval)', {
+      importModuleDynamically: common.mustNotCall(),
+    });
+
+    const result = s.runInContext(ctx);
+    await assert.rejects(result, {
+      code: 'ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING',
+    });
+  }
+
+  {
+    const m = new SourceTextModule('globalThis.fooResult = Promise.resolve("import(\'foo\')").then(eval)', {
+      context: ctx,
+      importModuleDynamically: common.mustNotCall(),
+    });
+    await m.link(common.mustNotCall());
+    await m.evaluate();
+
+    await assert.rejects(ctx.fooResult, {
+      code: 'ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING',
+    });
+    delete ctx.fooResult;
+  }
+}
+
+(async function() {
+  await test();
+  await testMissing();
+}()).then(common.mustCall());
```

Comment:
nit
```suggestion
Promise.all([
  test(),
  testMissing(),
]).then(common.mustCall());
```

Reply:
You could also rename the test to `.mjs` and use top-level `await`